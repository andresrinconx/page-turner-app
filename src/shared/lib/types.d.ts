import { COLORS } from "./constants";
import type { ViewStyle, TextStyle } from "react-native";

declare global {
  type FlexProps = {
    flex?: number;
    flexDirection?: ViewStyle["flexDirection"];
    justifyContent?: ViewStyle["justifyContent"];
    alignItems?: ViewStyle["alignItems"];
    flexWrap?: ViewStyle["flexWrap"];
    gap?: number;
  };

  type SpacingProps = {
    p?: number;
    px?: number;
    py?: number;
    pt?: number;
    pb?: number;
    pl?: number;
    pr?: number;
    m?: number;
    mx?: number;
    my?: number;
    mt?: number;
    mb?: number;
    ml?: number;
    mr?: number;
  };

  type PositionProps = {
    position?: ViewStyle["position"];
    top?: number;
    bottom?: number;
    left?: number;
    right?: number;
  };

  type SizingProps = {
    w?: number | string;
    h?: number | string;
    minW?: number | string;
    minH?: number | string;
    maxW?: number | string;
    maxH?: number | string;
  };

  type ColorProps = {
    bg?: keyof typeof COLORS;
    color?: keyof typeof COLORS;
  };

  type TypographyProps = {
    fontSize?: number;
    lineHeight?: number;
    fontWeight?: TextStyle["fontWeight"];
    textAlign?: TextStyle["textAlign"];
    fontFamily?: string;
    letterSpacing?: number;
    textDecorationLine?: TextStyle["textDecorationLine"];
    textTransform?: TextStyle["textTransform"];
  };

  type BorderProps = {
    borderWidth?: number;
    borderTopWidth?: number;
    borderBottomWidth?: number;
    borderLeftWidth?: number;
    borderRightWidth?: number;
    borderRadius?: number;
    borderTopLeftRadius?: number;
    borderTopRightRadius?: number;
    borderBottomLeftRadius?: number;
    borderBottomRightRadius?: number;
    borderStyle?: ViewStyle["borderStyle"];
    borderColor?: keyof typeof COLORS;
  };

  type CommonStyleProps = FlexProps &
    SpacingProps &
    SizingProps &
    PositionProps &
    ColorProps &
    TypographyProps &
    BorderProps;
}

export {};
