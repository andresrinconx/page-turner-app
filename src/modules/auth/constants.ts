/**
 * constants.ts
 * Constants for the auth module
 *
 * Created by Andres Rincon on 21/4/25.
 */
