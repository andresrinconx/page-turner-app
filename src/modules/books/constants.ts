/**
 * constants.ts
 * Constants for the books module
 *
 * Created by Andres Rincon on 21/4/25.
 */
