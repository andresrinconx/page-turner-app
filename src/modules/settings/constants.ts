/**
 * constants.ts
 * Settings module constants
 *
 * Created by Andres Rincon on 28/4/25.
 */

import { WeekdayItem } from "./types";

export const DEFAULT_WEEKDAYS: WeekdayItem[] = [
  { id: 1, label: "M", selected: true },
  { id: 2, label: "T", selected: true },
  { id: 3, label: "W", selected: true },
  { id: 4, label: "T", selected: true },
  { id: 5, label: "F", selected: true },
  { id: 6, label: "S", selected: true },
  { id: 7, label: "S", selected: true },
];

export const STORAGE_KEYS = {
  WEEKDAYS: "notifications_weekdays",
  START_DATE: "notifications_start_date",
  REMINDER_TIME: "notifications_reminder_time",
};

export const NOTIFICATION_DEFAULTS = {
  TITLE: "Page Turner",
  BODY: "Time to read your book today!",
  HOUR: 8,
  MINUTE: 0,
};
