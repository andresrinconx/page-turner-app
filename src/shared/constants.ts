/**
 * constants.ts
 * Constants for the shared components
 *
 * Created by Andres Rincon on 21/4/25.
 */

export const COLORS = {
  primary: "#0BB589",
  primaryLight: "#E4F7ED",
  secondary: "#EF844C",
  background: "#F7FBFC",
  error: "#B12B2C",
  success: "#219653",
  info: "#00CFE8",
  overlay: "#696969",
  text: "#5C605E",
  secondaryText: "#82868B",
  disabled: "#B9BBC3",
  progress: "#E0E4E5",
  inputText: "#EFEFEF",
  card: "#F7F9FD",
  white: "#FFFFFF",
  black: "#000000",
};

export const SIZES = {
  borderRadius: 16,
};
