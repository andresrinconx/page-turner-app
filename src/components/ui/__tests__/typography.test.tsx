import { render } from "@testing-library/react-native";
import Typography from "../typography";
import { COLORS } from "../../../lib/constants";

describe("Typography", () => {
  it("renders with default props", () => {
    const { getByText } = render(<Typography>Hello World</Typography>);
    const text = getByText("Hello World");
    expect(text).toBeTruthy();
    expect(text.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          fontSize: 16,
          lineHeight: 24,
        }),
      ]),
    );
  });

  it("renders with different variants", () => {
    const variants = [
      "paragraph1",
      "paragraph2",
      "paragraph3",
      "headline1",
      "headline2",
      "headline3",
      "headline4",
      "label1",
      "label2",
    ] as const;

    variants.forEach(variant => {
      const { getByText } = render(
        <Typography variant={variant}>Test Text</Typography>,
      );
      const text = getByText("Test Text");
      expect(text).toBeTruthy();
    });
  });

  it("applies primary and secondary types correctly", () => {
    const { getByText: getPrimary } = render(
      <Typography type="primary">Primary</Typography>,
    );
    const { getByText: getSecondary } = render(
      <Typography type="secondary">Secondary</Typography>,
    );

    const primaryText = getPrimary("Primary");
    const secondaryText = getSecondary("Secondary");

    expect(primaryText.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          color: COLORS.text,
        }),
      ]),
    );

    expect(secondaryText.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          color: COLORS.secondaryText,
        }),
      ]),
    );
  });

  it("applies different font weights", () => {
    const weights = ["light", "normal", "medium", "semibold", "bold"] as const;

    weights.forEach(weight => {
      const { getByText } = render(
        <Typography weight={weight}>Weight Test</Typography>,
      );
      const text = getByText("Weight Test");
      expect(text.props.style).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            fontWeight: expect.any(String),
          }),
        ]),
      );
    });
  });

  it("applies custom color", () => {
    const customColor = "#FF0000";
    const { getByText } = render(
      <Typography color={customColor}>Colored Text</Typography>,
    );
    const text = getByText("Colored Text");
    expect(text.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          color: customColor,
        }),
      ]),
    );
  });

  it("applies different text alignments", () => {
    const alignments = ["left", "center", "right"] as const;

    alignments.forEach(align => {
      const { getByText } = render(
        <Typography textAlign={align}>Aligned Text</Typography>,
      );
      const text = getByText("Aligned Text");
      expect(text.props.style).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            textAlign: align,
          }),
        ]),
      );
    });
  });

  it("applies custom styles", () => {
    const customStyle = { marginTop: 10 };
    const { getByText } = render(
      <Typography style={customStyle}>Styled Text</Typography>,
    );
    const text = getByText("Styled Text");
    expect(text.props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(customStyle)]),
    );
  });
});
