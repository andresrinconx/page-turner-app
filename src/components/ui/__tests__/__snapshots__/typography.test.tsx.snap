// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography matches snapshot with custom props 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 28,
        "lineHeight": 34,
      },
      {
        "textAlign": "center",
      },
      {
        "color": "#FF0000",
        "fontWeight": "700",
      },
      undefined,
    ]
  }
>
  Custom Snapshot
</Text>
`;

exports[`Typography matches snapshot with default props 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      {
        "textAlign": "left",
      },
      {
        "color": "#5C605E",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Snapshot Test
</Text>
`;
